### PR TITLE
Add https to metrics server

### PR DESCRIPTION
**Pull request by yuriykis**

### CODE REVIEW

File: tls-config.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+tls_server_config:
```
Comment (sysadmind) [inline tls-config.yml:1]: I don't think that this file is necessary.

File: main.go [hunk 2]
```diff
@@ -53,7 +54,8 @@ func main() {
 		metricsPath = kingpin.Flag("web.telemetry-path",
 			"Path under which to expose metrics.").
 			Default("/metrics").String()
-		esURI = kingpin.Flag("es.uri",
+		tlsConfigFile = kingpin.Flag("web.config", "Path to config yaml file that can enable TLS").Default("").String()
```
Comment (sysadmind) [inline main.go:57]: This should be handled by the kingpinflag module from the exporter-toolkit.
`webflag "github.com/prometheus/exporter-toolkit/web/kingpinflag"`

```suggestion
		webConfig = webflag.AddFlags(kingpin.CommandLine)
```

File: main.go [hunk 3]
```diff
@@ -288,11 +290,8 @@ func main() {
 	)
 
 	go func() {
-		if err := server.ListenAndServe(); err != nil {
-			_ = level.Error(logger).Log(
-				"msg", "http server quit",
-				"err", err,
-			)
+		if err := web.ListenAndServe(server, *tlsConfigFile, logger); err != nil {
```
Comment (sysadmind) [inline main.go:293]:
```suggestion
		if err := web.ListenAndServe(server, *webConfig, logger); err != nil {
```

**Review by sysadmind — CHANGES_REQUESTED** [review]
Just a small adjustment to the kingpin flags. Otherwise good. Thanks!

**Review by yuriykis — COMMENTED** [review]
Yes, the requested changes look good to me :)

**Review by sysadmind — APPROVED** [review]
LGTM. Thanks

**Pull request merged**